webapp/channels/src/packages/mattermost-redux/src/reducers/entities/channel_categories.ts

```diff
@@ -92,6 +95,44 @@ export function byId(state: IDMappedObjects<ChannelCategory> = {}, action: Gener
 
         return changed ? nextState : state;
     }
+    case ChannelTypes.RECEIVED_CHANNEL: {
+        const receivedChannel: Channel = toClientChannel(action.data);
+        if (receivedChannel.type === General.PRIVATE_CHANNEL) {
+            // For GM to Private channel conversion feature
+            // In the case when someone converts your GM to a private channel and moves it to a team
+            // you're not currently on, we need to remove the channel from "direct messages" category
+            // and add it to "channels" category of target team. Even though the server sends a websocket event about updated category data,
+            // it does so only for the team the channel got moved into, and not every team a user is part of, for performance reasons.
+            // For every other team, we update the state here. Everything is correct ion server side, but we update state here
+            // to avoid re-fetching all categories again for all teams.
+
+            const newState: IDMappedObjects<ChannelCategory> = {};
+            const categoryIDs = Object.keys(state);
+            categoryIDs.forEach((categoryID) => {
```

Comment:
I wonder if there is any way to simplify this, by passing somehow whether it is a conversion or not. Somewhere should be relatively simple to check if the type of the channel we had in redux was a GM and now is not.

Reply:
Done. In websocket event listener, I'm not checking if its a conversion by checking if channel exists in state as a GM and received channel is a private channel. If condition is met, I'm firing a new action, that only handles the task of fixing categories.